jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -216,6 +216,20 @@ private E linkNextConsumerChunkAndPoll(
         MpmcUnboundedXaddChunk<E> next,
         long expectedChunkIndex)
     {
+        if (next == null) {
+            final long cChunkIndex = expectedChunkIndex - 1;
+            assert cChunk.lvIndex() == cChunkIndex;
+            // no need to try more then once here:
+            // if there is already a rotation in place or
```

Comment:
I'm not sure what is guiding your line length here... Is this a poem? :P

Reply:
I admit it's a bit verbose :P
And I've read more inspired ones too, so I can rephrase it with  TLDR version :D

Reply:
mmm but you agree on the logic? There are other queues that can/should get same treatment? (Not the poem :P)